### PR TITLE
Updated GSVA module to include cancer_group comparison

### PR DESCRIPTION
**Pull request by sangeetashukla**
### Purpose/implementation Section
Update [gene-set-enrichment-analysis](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/gene-set-enrichment-analysis) module

#### What scientific question is your analysis addressing?
- Exclude `harmonized_diagnosis` and include `broad_histology` in addition to `cancer_group` and predictor variables
- Rectify incorrect naming of output files for harmonized diagnosis Turkey statistics 
- include only tumor samples for all cohorts
- exclude TCGA and GTex samples

#### What GitHub issue does your pull request address?
[Issue 455](https://github.com/PediatricOpenTargets/ticket-tracker/issues/455)

#### Is there anything that you want to discuss further?
No.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
gsva_anova_exome_capture_cancer_group.tsv
gsva_anova_polya_broad_histology.tsv
gsva_anova_polya_cancer_group.tsv
gsva_anova_polya_stranded_broad_histology.tsv
gsva_anova_polya_stranded_cancer_group.tsv
gsva_anova_stranded_broad_histology.tsv
gsva_anova_stranded_cancer_group.tsv
gsva_scores.tsv
gsva_tukey_exome_capture_cancer_group.tsv
gsva_tukey_polya_broad_histology.tsv
gsva_tukey_polya_cancer_group.tsv
gsva_tukey_polya_stranded_broad_histology.tsv
gsva_tukey_polya_stranded_cancer_group.tsv
gsva_tukey_stranded_broad_histology.tsv
gsva_tukey_stranded_cancer_group.tsv

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd [hunk 1]
```diff
@@ -162,17 +162,17 @@ for(i in 1:length(rna_library_list)){
   }
   
   # anova can only be run on factors with >=2 levels, so to avoid error, we give a if statement
-  if(harmonized_diagnosis_n>=2){
+  #if(harmonized_diagnosis_n>=2){
```
Comment (ewafula) [inline analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd]: @sangeetashukla,  Why are you removing >2 cutoffs?

File: analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd [hunk 2]
```diff
@@ -233,14 +233,14 @@ for(i in 1:length(rna_library_list)){
     pull(harmonized_diagnosis) %>% 
     unique() %>% length()
   
-  if(harmonized_diagnosis_n>=2){
+  #if(harmonized_diagnosis_n>=2){
```
Comment (ewafula) [inline analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd]: @sangeetashukla, Why are you removing >2 cutoffs?

File: analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd (hunk 1 above)
Comment (jharenza) [reply analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd]: To be honest, we can probably remove harm dx from this altogether....just keep broad hist and the cancer groups?

File: analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd (hunk 2 above)
Comment (sangeetashukla) [reply analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd]: @ewafula The issue description requires "Check to confirm that ... all harmonized diagnosis ANOVA statistics **don't meet** the >2 cutoffs". Am I missing something? On the other hand, as @jharenza suggests, I can go ahead and remove harm_dx altogether, keeping broad_hist and cancer_group.

**Comment by ewafula** [comment]
Check to confirm that missing results are due to not meeting the cutoff.
Not  to remove them.

There are also issues with the comparison column. You can hold on until I
complete reviewing.

**Review by ewafula — APPROVED** [review]
Thank you for @sangeetashukla working on this! 
@jharenza, I have updated the PR to exclude `harmonized_diagnosis` and include `broad_histology. 
Questions:
-  Should the module be updated to include other cohorts? Currently subsetting for  only PBTA
- PBTA's `broad_histology` column includes terms such as `Non-tumor` and `Other,` which are being used as a level factor for the predictor variable  (broad_histology in `ANOVA` and the post-hoc test, `TukeyHSD` tests. Should they be filtered out from histology files before the statistical tests are run?

**Comment by jharenza** [comment]
>Should the module be updated to include other cohorts? Currently subsetting for only PBTA

Yes, should do all here 

> PBTA's broad_histology column includes terms such as Non-tumor and Other, which are being used as a level factor for the predictor variable (broad_histology in ANOVA and the post-hoc test, TukeyHSD tests. Should they be filtered out from histology files before the statistical tests are run?

Yes, we can remove

**Comment by ewafula** [comment]
> > Should the module be updated to include other cohorts? Currently subsetting for only PBTA
> 
> Yes, should do all here
> 
> > PBTA's broad_histology column includes terms such as Non-tumor and Other, which are being used as a level factor for the predictor variable (broad_histology in ANOVA and the post-hoc test, TukeyHSD tests. Should they be filtered out from histology files before the statistical tests are run?
> 
> Yes, we can remove

@jharenza, I am assuming we need to filter for tumor samples only, which is currently not implemented in the code. Do we also include adult tumors in the TCGA cohort?

**Comment by jharenza** [comment]
>@jharenza, I am assuming we need to filter for tumor samples only, which is currently not implemented in the code. Do we also include adult tumors in the TCGA cohort?

Yes, tumor only (exclude gtex, should not have much normal RNA-Seq in other cohorts). We can leave TCGA out for now

**Pull request merged**